rfcs/20221124-monorepo.md

```diff
@@ -0,0 +1,178 @@
+
+An implementation of the GraphQL API call (albeit in Ruby) is available in `hub` and the [code for it](https://github.com/github/hub/commit/4c2e44146988dfb385a26f649298f274a5017756) is available in their GitHub repo for reference.
+
+However, instead of writing the code ourselves, we can install `hub` and run a small script that repeatedly calls `hub` to migrate each issue one by one. That'll be a hack but it's okay since this is a one-off use for this.
+
+### Step 2. Restore workflows
+
+The workflows of both the API and frontend will need some refactoring to start, and pass, again. [monopenverse](https://github.com/dhruvkb/monopenverse) has updated these workflows and the following work.
+
+[monopenverse](https://github.com/dhruvkb/monopenverse) showcases a `setup-env` action that sets up Node.js, Python, Just and other dependencies and can be used in every workflow.
+
+The `ci_cd.yml` workflow from the API has been very nicely combined with the `ci.yml` workflow from the frontend. Redundant steps were eliminated.
+
+The following actions have been successfully combined:
+
+- actionlint ✅
+- bundle_size.yml ✅
+- ci_cd.yml (API) + ci.yml (frontend) [merged] ✅
+- Playwright tests from ci.yml (frontend) ✅
+- draft_release.yml ✅
+- generate_pot.yml ✅
+- gh_pages.yml ✅
+- migration_safety_warning.yml ✅
+- subscribe_to_label.yml ✅
+- label_new_pr.yml ✅
+- pr_closed.yml ✅
+- pr_label_check.yml ✅
+- new_issues.yml ✅
+- pr_ping.yml ✅
+
+The following have not been verified to work:
+
+- renovate.yml
+- rollback.yml
+- ghcr.yml
+- push_docker_image.yml
+
+With this done, the development on the frontend can continue inside the subdirectory.
+
+### Step 3. Buff the rough edges
+
+There will be a few rough edges that I cannot foresee and we can continuously fix those as we spot them. But up to this point we should be in a position where
+we can continue to build the API and the frontend independently but from one repo.
+
+1. The action `banyan/auto-label` will need to be configured (`auto-label.json`) to add the "stack: \*" labels based on the modified directory.
+
+### Step 4. Integration
+
+This is the long term combination of code for the frontend and the API.
+
+#### Combined lint
+
+All lint steps can be combined in `.pre-commit-config.yaml`. This also simplifies the CI jobs can now be merged.
+
+See the combined lint in action at [monopenverse](https://github.com/dhruvkb/monopenverse).
+
+### Step 5. Documentation merge
+
+The following documentation files will need reorganisation or merge.
+
+- README.md (both repos)
+- CODE_OF_CONDUCT.md (both repos)
+- CONTRIBUTING.md (both repos)
+- CONTRIBUTORS.md (API only; also why?)
+- DOCUMENTATION_GUIDELINES.md (API only)
+- TESTING_GUIDELINES.md (frontend only)
+- DEPLOYMENT.md (frontend only)
+
+I will need more information about this because IANAL.
+
+- LICENSE (both repos)
```

Comment:
Given Openverse still uses MIT we have probably very little to worry about here, FWIW.

Reply:
Nice, but I had a vague feeling that merging repositories could have legal implications for those that use them somehow.